### PR TITLE
Handle new errors observed

### PR DESCRIPTION
**Pull request by WebSpider**
Add handling for 

- Failed authentications
- New terms and conditions
- Errors seen when service had problems

Depends on yet-to-be-released `myskoda` library with skodaconnect/myskoda#209 merged

### CODE REVIEW

File: custom_components/myskoda/__init__.py [hunk 1]
```diff
@@ -41,8 +49,17 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
 
     try:
         await myskoda.connect(config.data["email"], config.data["password"])
+    except AuthorizationFailedError:
+        _LOGGER.debug("Authorization with MySkoda failed.")
+        raise ConfigEntryAuthFailed("Authentication failed.")
+    except TermsAndConditionsError:
+        _LOGGER.debug("New terms and conditions detected in login flow.")
```
Comment (dvx76) [inline custom_components/myskoda/__init__.py:56]: Is that all the user gets, or does the re-raise make it pop up somewhere else? Do we want to be a bit more verbose here and provide instructions (_log out and back in again on the MySkoda app to access the new T&C_)?

Comment (dvx76) [reply custom_components/myskoda/__init__.py:56]: Also what determines whether we raise a new HA-specific exception, or re-raise the same exception, or return False?

**Review by dvx76 — APPROVED** [review]

File: custom_components/myskoda/__init__.py (hunk 1 above)
Comment (WebSpider) [reply custom_components/myskoda/__init__.py:56]: > Is that all the user gets, or does the re-raise make it pop up somewhere else? Do we want to be a bit more verbose here and provide instructions (_log out and back in again on the MySkoda app to access the new T&C_)?

I have not yet figured out how to make this a proper message in the integration status that also is translatable etc.

I will make the message a bit more elaborate, for instance adding an error level logger as well, while I figure out how I can properly inform the user in the UI as well.

Comment (WebSpider) [reply custom_components/myskoda/__init__.py:56]: > Also what determines whether we raise a new HA-specific exception, or re-raise the same exception, or return False?

You make it sounds like there is a determination there :grin:

I will make it a bit more consistent.

**Pull request merged**